src/lib/utils/date_time_utils.cpp

```diff
@@ -17,6 +18,21 @@ std::optional<boost::gregorian::date> string_to_date(const std::string& date_str
   return std::nullopt;
 }
 
+std::optional<boost::posix_time::ptime> string_to_date_time(const std::string& date_time_string) {
+  try {
+    const auto date_time = boost::posix_time::time_from_string(date_time_string);
+    if (!date_time.is_not_a_date_time()) {
+      return date_time;
+    }
+  } catch (const boost::wrapexcept<boost::gregorian::bad_day_of_month>&) {
```

Comment:
Add a comment why we ignore these issues and other not?

Reply:
Is `// We catch parsing errors since we return a std::nullopt if the input string is not a valid date/timestamp` sufficient?